mlir/include/mlir/Dialect/XeGPU/IR/XeGPUOps.td

```diff
@@ -163,7 +174,26 @@ def XeGPU_CreateNdDescOp: XeGPU_Op<"create_nd_tdesc", [Pure, ViewLikeOpInterface
     }
 
     ArrayRef<int64_t> getStaticOffsets(){
-      return getConstOffsets();
+      auto attr = getConstOffsetsAttr();
+
+      if (attr) 
+        return attr;
+
+      auto memrefType = llvm::dyn_cast<MemRefType>(getSourceType());
+      int rank = 0;
+      if (memrefType) 
+        //use source memref's rank, as source memref rank may be higher
+        rank = memrefType.getRank();
+      else
```

Comment:
nit: use braces as comments make it a multi-line body (if you need to keep these blocks at all)